File: solutions/allergies/C++/allergies.cpp
#include "allergies.h"

namespace allergies {
    allergy_test::allergy_test(int score) {
        const std::unordered_map<std::string, int> allergen_scores({
            {"eggs",         1},
            {"peanuts",      2},
            {"shellfish",    4},
            {"strawberries", 8},
            {"tomatoes",     16},
            {"chocolate",    32},
            {"pollen",       64},
            {"cats",         128}
        });

        for (auto x : allergen_scores) {
            if ((score & x.second) == x.second)
                allergens.emplace(x.first);
        }
    }

    bool allergy_test::is_allergic_to(std::string str) {
        return allergens.count(str);
    }

    std::unordered_set<std::string> allergy_test::get_allergies() {
        return allergens;
    }

}

File: solutions/allergies/Python/allergies.py
class Allergies:
    allergens = ["eggs", "peanuts", "shellfish", "strawberries", "tomatoes", "chocolate", "pollen", "cats"]
    
    def __init__(self, score):
        self.score = score
        self.allergies = [x for i, x in enumerate(self.allergens) if score == score | 1 << i]

    def allergic_to(self, item):
        return item in self.allergies

    @property
    def lst(self):
        return [name for name in self.allergens if self.allergic_to(name)]
